x-pack/plugins/security/server/authentication/http_authentication/http_authorization_header.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { KibanaRequest } from '../../../../../../src/core/server';
+
+export class HTTPAuthorizationHeader {
+  constructor(public scheme: string, public credentials: string) {}
```

Comment:
__nit:__ (assuming we don't want to modify this down the road).
```suggestion
  constructor(public readonly scheme: string, public readonly credentials: string) {}
```

Reply:
Good call, let's make this `readonly` until it doesn't make sense to anymore. I can't think of a situation where we'd need to be changing this.